nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -374,7 +396,16 @@ void CollisionMonitor::process(const Velocity & cmd_vel_in)
   // Fill collision_points array from different data sources
   for (std::shared_ptr<Source> source : sources_) {
     if (source->getEnabled()) {
-      source->getData(curr_time, collision_points);
+      if (!source->getData(curr_time, collision_points)) {
+        RCLCPP_WARN(get_logger(), "Invalid blocking source detected, stopping the robot");
```

Comment:
Would be useful to know which source is faulty

Reply:
See last commit